dev-cpp/functional-plus/functional-plus-0.2.22.ebuild

```diff
@@ -0,0 +1,35 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake
+
+if [[ ${PV} == *9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/Dobiasd/FunctionalPlus.git"
+else
+	SRC_URI="https://github.com/Dobiasd/FunctionalPlus/archive/refs/tags/v${PV}.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64"
+fi
+
+DESCRIPTION="Functional Programming Library for C++"
+HOMEPAGE="https://www.editgym.com/fplus-api-search/"
+
+LICENSE="Boost-1.0"
+SLOT="0"
+
+IUSE="test"
+RESTRICT="!test? ( test )"
+
+DEPEND="test? ( dev-cpp/doctest )"
+
+S="${WORKDIR}/FunctionalPlus-${PV}"
+
+src_test() {
+	local BUILD_DIR="${WORKDIR}/${P}_build/test"
+	local CMAKE_USE_DIR="${S}/test"
+	cmake_src_configure
```

Comment:
Not a fan of trying to do this in src_test. Can you conditionally do it in src_configure/src_compile instead?

Reply:
Done! It looks much better now :)